### ISSUE TITLE
ConsecutiveIndex constraint fail when using multiple Apply methods on save entity

### ISSUE THREAD

**Issue by vermilion** [Finaps/EventSourcing#72]
**Current behavior:**
When using multiple `Apply` methods `EFCore` provider doesn't preserve their order on insert. 

My code that fails
```
        Apply(new CompetitionCreatedEvent());
        Apply(new CompetitionNameSetEvent(name));
        Apply(new CompetitionLogoSetEvent(logo));
```
EF then try to sort entities alphabetically on insert thus getting into constraint violation 

You can check issue here [https://github.com/dotnet/efcore/issues/11686](https://github.com/dotnet/efcore/issues/11686)
This issue lead me to PR [https://github.com/dotnet/efcore/pull/25874](https://github.com/dotnet/efcore/pull/25874) where inserted entities are ordered using `StringComparer.Ordinal`

After digging source code found the piece of code that cause this
```
	case AddEventsAction(var events):
    	_store.Context.AddRange(events);
        break;
```

**Proposed solution:**
As a solution I can only think of `Bulk insert` wrapped in transaction. Maybe you'll find a better way

**Comment by BramKraai** [comment]
Hey,

Thanks for your very well written issue! 

We were able to reproduce the issue in our tests and saw that EF Core indeed does not preserve insertion order, leading to a violated ```ConsecutiveIndex``` constraint while using Postgres. For reasons unknown to me, the same error does not occur in SQL Server.

I've linked a proposed solution PR, but we might investigate alternative solutions before merging the PR.
